dhis-2/dhis-api/src/main/java/org/hisp/dhis/user/User.java

```diff
@@ -1024,27 +1006,15 @@ public boolean isAuthorized( Authorities auth )
 
     public Set<UserGroup> getManagedGroups()
     {
-        Set<UserGroup> managedGroups = new HashSet<>();
-
-        for ( UserGroup group : groups )
-        {
-            managedGroups.addAll( group.getManagedGroups() );
-        }
-
-        return managedGroups;
+        return groups == null ? Set.of()
+            : groups.stream()
+                .flatMap( group -> emptyIfNull( group.getManagedGroups() ).stream() )
+                .collect( Collectors.toUnmodifiableSet() );
     }
 
     public boolean hasManagedGroups()
     {
-        for ( UserGroup group : groups )
-        {
-            if ( group != null && group.getManagedGroups() != null && !group.getManagedGroups().isEmpty() )
-            {
-                return true;
-            }
-        }
-
-        return false;
+        return getManagedGroups().stream().anyMatch( Objects::nonNull );
```

Comment:
You misunderstood my intention in the previous comment, so the idea was not to call `getManagedGroups()` to avoid building intermediate collection(s), instead you would do basically the same first part of the code, but then use `anyMatch` to see if there is something to find.
```java
        return groups == null ? false
            : groups.stream()
                .anyMatch(group -> group.getManagedGroups() != null && !group.getManagedGroups().isEmpty);
```
Similar with `hasAuthorities()` ;)